docs/02.API-REFERENCE.md

```diff
@@ -4692,6 +4767,69 @@ jerry_create_typedarray_for_arraybuffer_sz (jerry_typedarray_type_t type_name,
 - [jerry_release_value](#jerry_release_value)
 
 
+## jerry_create_container
+
+**Summary**
+
+Create a jerry_value_t representing a given type container object.
+
+*Notes*:
+- Returned value must be freed with [jerry_release_value](#jerry_release_value)
+  when it is no longer needed.
+- This API depends on a build option (`JERRY_ES2015_BUILTIN_CONTAINER`) and can be checked
+  in runtime with the `JERRY_FEATURE_MAP, JERRY_FEATURE_SET, JERRY_FEATURE_WEAKMAP, JERRY_FEATURE_WEAKSET`
+  feature enum values.
+  see: [jerry_is_feature_enabled](#jerry_is_feature_enabled).
+- The ES2015-subset profile enables this by default.
+
+*New in version 2.3*.
+
+**Prototype**
+
+```c
+jerry_value_t
+jerry_create_container (jerry_container_type_t container_type,
+                        const jerry_value_t *arguments_list_p,
+                        jerry_length_t arguments_list_len);
+```
+
+- `container_type` - Type of the container to be created, see `jerry_container_type_t`.
+- `arguments_list_p` - The arguments passed to the container constructor to be inserted to the container.
+- `arguments_list_len` - The length of the above arguments.
+- return value - the new container object as a `jerry_value_t`
+
+**Example**
+
+[doctest]: # ()
+
+```c
+#include "jerryscript.h"
+
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+  jerry_char_t src[] = "[1,2,3,4].entries()";
+  jerry_value_t iterable = jerry_eval (src, sizeof (src) - 1, JERRY_PARSE_NO_OPTS);
+
+  jerry_value_t map = jerry_create_container (JERRY_CONTAINER_TYPE_MAP, &iterable, 1);
```

Comment:
What would happen if I pass an js array to the argument list?

Reply:
The same thing that would happen if you passed an Array to the new Map() constructor in JavaScript. It would throw a TypeError.
You can expect the API to work the same as the JavaScript, since it only wraps the builtin routines.

Reply:
I see, then it would be good to describe this in the docs above. That this method will work the same way as the JS constructor calls.

Reply:
Good idea!